x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -177,25 +179,18 @@ public void getSnapshotInfo(GetSnapshotInfoContext context) {
         assert snapshotIds.size() == 1 && SNAPSHOT_ID.equals(snapshotIds.iterator().next())
             : "RemoteClusterRepository only supports " + SNAPSHOT_ID + " as the SnapshotId but saw " + snapshotIds;
         try {
-            getRemoteClusterClient().admin()
-                .cluster()
-                .prepareState()
-                .clear()
-                .setMetadata(true)
-                .setNodes(true)
-                // fork to the snapshot meta pool because the context expects to run on it and asserts that it does
-                .execute(new ThreadedActionListener<>(logger, threadPool, ThreadPool.Names.SNAPSHOT_META, context.map(response -> {
-                    Metadata responseMetadata = response.getState().metadata();
-                    Map<String, IndexMetadata> indicesMap = responseMetadata.indices();
-                    return new SnapshotInfo(
-                        new Snapshot(this.metadata.name(), SNAPSHOT_ID),
-                        List.copyOf(indicesMap.keySet()),
-                        List.copyOf(responseMetadata.dataStreams().keySet()),
-                        List.of(),
-                        response.getState().getNodes().getMaxNodeVersion(),
-                        SnapshotState.SUCCESS
-                    );
-                }), false));
+            getRemoteState(new ThreadedActionListener<>(logger, threadPool, ThreadPool.Names.SNAPSHOT_META, context.map(response -> {
+                Metadata responseMetadata = response.metadata();
+                Map<String, IndexMetadata> indicesMap = responseMetadata.indices();
+                return new SnapshotInfo(
+                    new Snapshot(this.metadata.name(), SNAPSHOT_ID),
+                    List.copyOf(indicesMap.keySet()),
+                    List.copyOf(responseMetadata.dataStreams().keySet()),
+                    List.of(),
+                    response.getNodes().getMaxNodeVersion(),
+                    SnapshotState.SUCCESS
+                );
+            }), false));
```

Comment:
Should we now add `assert false` to the exception block below? Seems like no exceptions should occur anymore, since `getRemoteState` handles it's own exceptions. If it does, we may have double invoked the listener.

Reply:
++  added